tests/publiccloud/migration.pm

```diff
@@ -38,12 +38,58 @@ sub run {
 
     sleep 90;    # wait for a bit for zypper to be available
 
-    record_info('installs SLE15-Migration and suse-migration-sle15-activation');
+    #Download unpublished DMS package and install rpm
+    my $url = get_var('PUBLIC_CLOUD_DMS_IMAGE_LOCATION');
+    my $package = get_var('PUBLIC_CLOUD_DMS_PACKAGE');
+    my $pc_url = get_var('PUBLIC_CLOUD_DMS_PC_LOCATION');
+    my $pc_pkg = get_var('PUBLIC_CLOUD_DMS_PC_PACKAGE');
+    my $act_url = get_var('PUBLIC_CLOUD_DMS_ACT_LOCATION');
+    my $act_pkg = get_var('PUBLIC_CLOUD_DMS_ACT_PACKAGE');
+    my $dms_repo_key_url = get_var('PUBLIC_CLOUD_DMS_REPO_KEY_URL');
+    my $dms_repo_key = get_var('PUBLIC_CLOUD_DMS_REPO_KEY');
+    my $dms_rpm = "$url"."$package";
+    my $act_rpm = 'N/A';
+    my $pc_rpm = 'N/A';
+    my ($v1,$dms_rpm_name) = split /\//,$package;
+    my $act_out = `curl -s $act_url`;
+    $act_rpm = $1 if $act_out =~ /($act_pkg.*?rpm)/g;
+    my $pc_out = `curl -s $pc_url`;
+    $pc_rpm = $1 if $pc_out =~ /($pc_pkg.*?rpm)/g;
+    my $dms_act_rpm = "$act_url"."$act_rpm";
+    my $dms_pc_rpm = "$pc_url"."$pc_rpm";
+
+    record_info("dms package name", $dms_rpm_name);
+    record_info("act package name", $act_rpm);
+    record_info("pc package name", $pc_rpm);
+
+    record_info('wget repokey and rpm');
+    assert_script_run("wget $dms_repo_key_url -O /tmp/$dms_repo_key", 180);
+    assert_script_run("wget $dms_rpm -O /tmp/$dms_rpm_name", 180);
+    assert_script_run("wget $dms_act_rpm -O /tmp/$act_rpm", 180);
+    assert_script_run("wget $dms_pc_rpm -O /tmp/$pc_rpm", 180);
+
+    record_info('SCP repokey and rpm');
+    #Download scp the rpm packages and repokey to public cloud
+    my $repo_key = $instance->scp("/tmp/$dms_repo_key", 'remote:' . "/tmp/$dms_repo_key", 200);
+    my $dms_rpm = $instance->scp("/tmp/$dms_rpm_name", 'remote:' . "/tmp/$dms_rpm_name", 1000);
+    my $dms_act_rpm = $instance->scp("/tmp/$act_rpm", 'remote:' . "/tmp/$act_rpm", 1000);
+    my $dms_pc_rpm = $instance->scp("/tmp/$pc_rpm", 'remote:' . "/tmp/$pc_rpm", 1000);
+
 
     # Upload distro_migration.log
     $instance->upload_log("/system-root/var/log/distro_migration.log", failok => 1);
 
-    $instance->ssh_script_run(cmd => "sudo zypper -n in SLES15-Migration suse-migration-sle15-activation", timeout => 300);
+    record_info('installs SLE15-Migration and suse-migration-sle15-activation');
+    #Download scp the rpm packages and repokey to public cloud
+    $instance->run_ssh_command(cmd => "sudo rpm --import /tmp/$dms_repo_key", proceed_on_failure => 0);
+    $instance->run_ssh_command(cmd => "sudo zypper in -y /tmp/$pc_rpm", proceed_on_failure => 0);
+    $instance->run_ssh_command(cmd => "sudo zypper in -y /tmp/$dms_rpm_name", proceed_on_failure => 0);
```

Comment:
I get lost in your code can you put expected name of the package which you installing here ?